corehq/apps/email/models.py

```diff
@@ -0,0 +1,153 @@
+import jsonfield
+from django.db import models
+
+from corehq.apps.email.util import BadEmailConfigException, get_email_backend_classes
+from corehq.util.quickcache import quickcache
+
+
+class SQLEmailSMTPBackend(models.Model):
+    objects = models.Manager()
+
+    # This tells us which type of backend this is
+    hq_api_id = models.CharField(max_length=126, null=True)
+
+    # This is the domain that the backend belongs to
+    domain = models.CharField(max_length=126, null=True, db_index=True)
+
+    # Name of the backend - e.g. "aws"
+    name = models.CharField(max_length=126, null=True)
+
+    # Simple name to display to users - e.g. "AWS"
+    display_name = models.CharField(max_length=126, null=True)
+
+    # Optionally, a description of this backend
+    description = models.TextField(null=True)
+
+    username = models.CharField(max_length=100)
+
+    password = models.CharField(max_length=150)
+
+    server = models.CharField(max_length=50)
+
+    port = models.CharField(max_length=5)
+
+    extra_fields = jsonfield.JSONField(default=dict)
+
+    is_default = models.BooleanField(default=False)
+
+    class Meta(object):
+        db_table = 'messaging_emailbackend'
+        app_label = 'email'
+
+    def __str__(self):
+        return "Domain '%s' Backend '%s'" % (self.domain, self.name)
+
+    def domain_is_authorized(self, domain):
+        """
+        Returns True if the given domain is authorized to use this backend.
+        """
+        return domain == self.domain
+
+    @classmethod
+    def name_is_unique(cls, name, domain, backend_id=None):
```

Comment:
I don't see a reference where backend_id is optional, so may be it can be an arg instead of a kwarg?

Reply:
Done